File: javascript/download-artifact.mjs
const API_BASE = process.env.TRANSCRIBETOOL_BASE_URL || "https://transcribetool.com";
const API_KEY = process.env.TRANSCRIBETOOL_API_KEY;
const transcriptId = process.argv[2];
const kind = process.argv[3] || "transcript_md";

if (!API_KEY) throw new Error("Set TRANSCRIBETOOL_API_KEY");
if (!transcriptId) throw new Error("Usage: node javascript/download-artifact.mjs <transcriptId> [kind]");

const response = await fetch(`${API_BASE}/api/v1/transcripts/${transcriptId}/downloads/${kind}`, {
  headers: { "x-api-key": API_KEY },
});

if (!response.ok) {
  console.error(await response.text());
  process.exit(1);
}

const text = await response.text();
console.log(text);

File: python/download_artifact.py
import os
import sys
import urllib.request

API_BASE = os.environ.get("TRANSCRIBETOOL_BASE_URL", "https://transcribetool.com")
API_KEY = os.environ.get("TRANSCRIBETOOL_API_KEY")
TRANSCRIPT_ID = sys.argv[1] if len(sys.argv) > 1 else None
KIND = sys.argv[2] if len(sys.argv) > 2 else "transcript_md"

if not API_KEY:
    raise RuntimeError("Set TRANSCRIBETOOL_API_KEY")
if not TRANSCRIPT_ID:
    raise RuntimeError("Usage: python3 python/download_artifact.py <transcript_id> [kind]")

req = urllib.request.Request(
    f"{API_BASE}/api/v1/transcripts/{TRANSCRIPT_ID}/downloads/{KIND}",
    headers={"x-api-key": API_KEY},
    method="GET",
)

with urllib.request.urlopen(req) as resp:
    print(resp.read().decode("utf-8"))
